test/test_main.py

```diff
@@ -0,0 +1,251 @@
+    def setUp(self):
+        self.config = get_config(None) # get user specific config
+        self.output_dir = tempfile.mkdtemp()
+        self.summary_fp = os.path.join(self.output_dir, "summary.txt")
+
+    def tearDown(self):
+        shutil.rmtree(self.output_dir)
+
+    def run_pipeline(self, r1, r2, config_file):
+
+        args = [
+            "--forward-reads", r1.name,
+            "--reverse-reads", r2.name,
+            "--summary-file", self.summary_fp,
+            "--output-dir", self.output_dir,
+            "--config-file", config_file.name
+            ]
+
+        main(args)
+
+    def test_main_RAP_bestHit(self):
+        #generate the correct config file
+        self.config['mapping_method'] = 'best_hit'
+        config_file = tempfile.NamedTemporaryFile(suffix=".json")
+        json.dump(self.config, config_file)
+        config_file.seek(0)
+        
+        # create the fake input file
+        r1 = tempfile.NamedTemporaryFile(suffix=".fastq")
+        r1.write(MOCK_R1)
+        r1.seek(0)
+        r2 = tempfile.NamedTemporaryFile(suffix=".fastq")
+        r2.write(MOCK_R2)
+        r2.seek(0)
+
+        # run the pipeline
+        self.run_pipeline(r1, r2, config_file)
+
+        # check if the output is correct
+        myoutput_fp = os.path.join(self.output_dir, os.path.basename(os.path.splitext(r1.name)[0]+'.ko'))
+        observed = open(myoutput_fp).read()
+        self.assertEqual(observed, EXPECTED_OUTPUT)
+        
+        # check if the summary file is correct
+        with open(self.summary_fp) as f:
+            observed = json.load(f)
+        self.assertEqual(observed.get('data', {}), EXPECTED_SUMMARY)
+
+    def test_make_index_bestHit(self):
+        # create the mock kegg file
+        kegg = tempfile.NamedTemporaryFile()
+        kegg.write(MOCK_KEGG)
+        kegg.seek(0)
+        
+        # create temp directory for the index
+        kegg2ko = tempfile.NamedTemporaryFile()
+        self.config['kegg_fp'] = kegg.name
+        self.config['kegg_to_ko_fp'] = kegg2ko.name
+
+        # make index
+        assignerApp = Assignment(self.config)
+        assignerApp.make_index()
+
+        # check equal
+        observed = open(self.config["kegg_to_ko_fp"]).read()
+        self.assertEqual(observed.strip(), KEGG_TO_KO.strip())
+
+    def test_empty_alignment(self):
+        config_file = tempfile.NamedTemporaryFile(suffix=".json")
+        json.dump(self.config, config_file)
+        config_file.seek(0)
+        
+        r1 = tempfile.NamedTemporaryFile(suffix=".fastq")
+        r1.write(MOCK_NO_MATCH_R1)
+        r1.seek(0)
+        r2 = tempfile.NamedTemporaryFile(suffix=".fastq")
+        r2.write(MOCK_NO_MATCH_R2)
+        r2.seek(0)
+                                
+        self.run_pipeline(r1, r2, config_file)
+        
+        #check if the output is correct
+        myoutput_fp = os.path.join(self.output_dir, os.path.basename(os.path.splitext(r1.name)[0]+'.ko'))
```

Comment:
This is repeated several times.  I wonder if we can get this logic into the code rather than keeping it in the test.